src/Workspaces/VisualBasic/Portable/Editing/VisualBasicImportAdder.vb

```diff
@@ -0,0 +1,90 @@
+
+Imports System.Collections.Immutable
+Imports System.Composition
+Imports System.Threading
+Imports Microsoft.CodeAnalysis
+Imports Microsoft.CodeAnalysis.Editing
+Imports Microsoft.CodeAnalysis.Host.Mef
+Imports Microsoft.CodeAnalysis.Options
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+Imports Microsoft.CodeAnalysis.VisualBasic.Utilities
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.Editing
+    <ExportLanguageService(GetType(ImportAdderService), LanguageNames.VisualBasic), [Shared]>
+    Partial Friend Class VisualBasicImportAdder
+        Inherits ImportAdderService
+
+        Protected Overrides Sub GetExistingImportedNamespaces(document As Document, model As SemanticModel, namespaces As HashSet(Of INamespaceSymbol))
+            namespaces.AddRange(model.Compilation.MemberImports.OfType(Of INamespaceSymbol))
+
+            ' consider all imports clauses
+            Dim root = DirectCast(model.SyntaxTree.GetRoot(), CompilationUnitSyntax)
+            For Each import In root.Imports
+                For Each clause In import.ImportsClauses
+                    Dim symbol = GetImportedNamespaceSymbol(clause, model)
+                    If symbol IsNot Nothing Then
+                        namespaces.Add(symbol)
+                    End If
+                Next
+            Next
+        End Sub
+
+        Protected Overrides Function GetImportedNamespaceSymbol(namespaceImport As SyntaxNode, model As SemanticModel) As INamespaceSymbol
+            Select Case namespaceImport.Kind
+                Case SyntaxKind.ImportsStatement
+                    Dim import = DirectCast(namespaceImport, ImportsStatementSyntax)
+                    Return GetImportedNamespaceSymbol(import.ImportsClauses(0), model)
+                Case SyntaxKind.SimpleImportsClause
+                    Dim clause = DirectCast(namespaceImport, SimpleImportsClauseSyntax)
+                    Return TryCast(model.GetSymbolInfo(clause.Name).Symbol, INamespaceSymbol)
+                Case Else
+                    Return Nothing
+            End Select
+        End Function
+
+        Protected Overrides Function GetExplicitNamespaceSymbol(node As SyntaxNode, model As SemanticModel) As INamespaceSymbol
+            Dim qname = TryCast(node, QualifiedNameSyntax)
+            If qname IsNot Nothing Then
+                Return GetExplicitNamespaceSymbol(qname, qname.Left, model)
+            End If
+
+            Dim maccess = TryCast(node, MemberAccessExpressionSyntax)
+            If maccess IsNot Nothing Then
+                Return GetExplicitNamespaceSymbol(maccess, maccess.Expression, model)
+            End If
+
+            Return Nothing
+        End Function
+
+        Private Overloads Function GetExplicitNamespaceSymbol(fullName As ExpressionSyntax, namespacePart As ExpressionSyntax, model As SemanticModel) As INamespaceSymbol
+            ' name must refer to something that Is Not a namespace, but be qualified with a namespace.
```

Comment:
'Is Not' probably wasn't intended to be capitalized. Pretty-lister does it when you copy a comment from C#.